sky/resources.py

```diff
@@ -1123,6 +1221,45 @@ def _try_validate_disk_tier(self) -> None:
                         f'Disk tier {self.disk_tier.value} is not supported '
                         f'for instance type {self.instance_type}.') from None
 
+    def _try_validate_volumes(self) -> None:
+        """Try to validate the volumes attribute.
+
+        Raises:
+            ValueError: if the attribute is invalid.
+        """
+        if self.volumes is None:
+            return
+        if self.cloud is None:
+            return
```

Comment:
Shall we raise an error here? The cloud must be gcp when volumes is not None I think

Reply:
Updated